src/YAMLValidator/Utility.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using YamlDotNet.Core;
+using YamlDotNet.RepresentationModel;
+
+namespace Microsoft.PowerPlatform.PowerApps.Persistence;
+internal sealed class YamlValidatorUtility
+{
+    public static string ReadFileData(string filePath)
+    {
+        try
+        {
+            using var reader = new StreamReader(filePath);
```

Comment:
FYI: you can use [File.ReadAll](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.readalltext?view=net-8.0)